janitor/functions.py

```diff
@@ -1575,6 +1571,52 @@ def collapse_levels(df: pd.DataFrame, sep: str = "_"):
     return df
 
 
+@pf.register_dataframe_method
+def reset_index_inplace(df: pd.DataFrame, *args, **kwargs):
+    """
+    In Pandas, `reset_index()`, when used in place, does not return a
```

Comment:
I think it's also important to emphasize why this exists over doing method-chained `.reset_index()`. Memory usage, IIRC?

Please let me know what you think about it.

```suggestion
    Returns the dataframe with an inplace resetting of the index.
    
    Compared to non-inplace resetting, this avoids data copying, thus
    providing a potential speedup.
 
    In Pandas, `reset_index()`, when used in place, does not return a
```

Reply:
Both memory and compute time, yeah. Looks good.